web/ui/react-app/src/pages/targets/ScrapePoolContent.tsx

```diff
@@ -62,6 +72,50 @@ const ScrapePoolContentTable: FC<InfiniteScrollItemsProps<Target>> = ({ items })
   );
 };
 
+const stringColumns = ['scrapeUrl', 'health', 'lastError', 'lastScrape'] as const;
+type StringColumn = typeof stringColumns[number];
+const numericColumns = ['lastScrapeDuration'] as const;
+type NumericColumn = typeof numericColumns[number];
+type SortableColumn = StringColumn | NumericColumn;
+
+const compareFunc = (sortState: SortState) => (a: Target, b: Target) => {
+  // We need to distinguish between string columns and number columns to handle
+  // sorting appropriately.
+  if (stringColumns.includes(sortState.column as StringColumn)) {
```

Comment:
I would love some input here, i.e. how to remove the casting. I was thinking to do some type narrowing with a `SortState.type: "numeric" | "string"` but not sure if that's overkill or if there's a more obvious way.